src/realm/alloc_slab.cpp

```diff
@@ -774,6 +775,11 @@ ref_type SlabAlloc::attach_file(const std::string& path, Config& cfg)
     try {
         // we'll read header and (potentially) footer
         File::Map<char> map_header(m_file, File::access_ReadOnly, sizeof(Header));
+#if REALM_ENABLE_ENCRYPTION
+        if (auto enc = map_header.get_encrypted_mapping()) {
+            enc->set_observer(m_write_observer);
+        }
+#endif
```

Comment:
Should the WriteObserver just be a parameter to the Map constructor? Could consolidate all these `#if` to one place.

Reply:
good idea